packages/location-icon/src/index.tsx

```diff
@@ -1,13 +1,36 @@
-import React from "react";
-import PropTypes from "prop-types";
+import React, { ReactElement } from "react";
 
 import * as Styled from "./styled";
 
+type Props = {
+  className?: string;
+  /**
+   * Can be either a number or a string.
+   * See https://github.com/jacobwgillespie/styled-icons#props
+   */
+  size: number | string;
+  /**
+   * Title as used by styled-icons. If left blank defaults to either
+   * `From Location Icon` or `To Location Icon`.
+   * See https://github.com/jacobwgillespie/styled-icons#props
+   */
+  title?: string;
+  /**
+   * `from` or `to` or some other string value to trigger generic place icon.
+   */
+  type: string;
+};
+
 /**
  * LocationIcon provides a consistent icon for rendering from, to, or generic
  * place icons in form components like LocationField and in map overlays/popups.
  */
-const LocationIcon = ({ className, size, title, type }) => {
+export function LocationIcon({
+  className = "",
+  size = 10,
+  title = "",
+  type = ""
+}: Props): ReactElement {
```

Comment:
In the end, I prefer this syntax with the type declaration much more than prop-types!